<launch>
	<arg name="enableShadders" value="true" />

	<!-- Launch file for UDG - Valve Turning Scenario -->
	
	<!-- Load G500 components -->
 	<include file="$(find udg_pandora)/launch/start_g500-sim-Pandora.launch" />
	
	<!-- Run UWSim -->
	<node if="$(arg enableShadders)" name="UWSim" pkg="UWSim" type="UWSim" args=" --configfile $(find udg_pandora)/scenes/cirsPandora.xml --dataPath $(find udg_pandora)/" respawn="false"/>
	<node unless="$(arg enableShadders)" name="UWSim" pkg="UWSim" type="UWSim" args="--disableShaders --configfile $(find udg_pandora)/scenes/cirsPandora.xml --dataPath $(find udg_pandora)/" respawn="false"/>

	<!-- Load meta-data -->
	<rosparam command="load" file="$(find udg_pandora)/meta_data/visual_detector_meta_data.yaml" />
	<rosparam command="load" file="$(find udg_pandora)/meta_data/acoustic_detector_meta_data.yaml" />
	
	<!-- Load config files -->
	<rosparam command="load" file="$(find udg_pandora)/config/visual_detector.yaml" />

	<!-- Load nodes -->
	<node name="visual_detector" pkg="udg_pandora" type="visual_detector.py"/>
	<node name="acoustic_detector" pkg="udg_pandora" type="acoustic_detector.py"/>
	<node name="phdslam" pkg="udg_pandora" type="g500slam.py" />

	<!-- stereo image proc -->	
	<node name="image_proc" pkg="stereo_image_proc" type="stereo_image_proc">
    		<env name="ROS_NAMESPACE" value="/stereo_camera"/>
  	</node>
	
</launch>

